### ISSUE TITLE
leaks signal listeners in Node.js

### ISSUE THREAD

**Issue by boneskull** [google/zx#881]
### Expected Behavior

Given `test.mjs`

```js
import {$} from 'zx';

const ac = new AbortController();
const {signal} = ac;

for (let i = 0; i < 10; i++) {
  $.sync({signal})``; // does nothing
  // $.sync({signal})`pwd`; // same result
}

// no MaxListenersExceededWarning
```

I would expect the signal listener to be released after the subprocess has ended.

### Actual Behavior

Executing `node test.mjs` results in:

```
(node:49248) MaxListenersExceededWarning: Possible EventTarget memory leak detected. 11 abort listeners
added to [AbortSignal]. MaxListeners is 10. Use events.setMaxListeners() to increase limit
```

Not specific to `sync`.

### Steps to Reproduce

1. Create `test.mjs` with the example code
2. Run it via `node --trace-warnings test.mjs`
3. Observe a lovely stack trace blaming `zx/build/vendor-core.cjs:404:25`

### Specifications

- zx version: v8.1.5
- Platform: macos
- Runtime: node

**Comment by antongolub** [comment]
@boneskull,

Could you check [8.1.6-dev.5d2ad78](https://www.npmjs.com/package/zx/v/8.1.6-dev.5d2ad78)?

**Comment by boneskull** [comment]
@antongolub 

It appears that this only fixes the `sync` case.  This still emits warnings:

```js
import {$} from 'zx';

const ac = new AbortController();
const {signal} = ac;

for (let i = 0; i < 10; i++) {
  await $({signal})``; // does nothing
}
```

**Comment by boneskull** [comment]
I'm not sure if it fixes the `sync` case either.  [c575cd4](https://github.com/webpod/zurk/commit/c575cd4be9cabc3aa0821d15bf8f6fa08b7d299e) removes the listener _if the signal is aborted_, but nothing is removing the listener if the signal is not aborted afaict.

**Comment by antongolub** [comment]
@boneskull

Thanks for highlighting the issue. Everything should be finally fixed in [v8.1.6](https://github.com/google/zx/releases/tag/8.1.6)

**Comment by boneskull** [comment]
@antonmedv I can confirm it's fixed. Thank you!
